src/Bicep.Core/Emit/ExpressionEmitter.cs

```diff
@@ -99,6 +99,16 @@ public void EmitUnqualifiedResourceId(DeclaredResourceMetadata resource, IndexRe
             writer.WriteValue(serialized);
         }
 
+        public void EmitFullyQualifiedResourceId(DeclaredResourceMetadata resource, IndexReplacementContext? indexContext)
+        {
+            var converterForContext = converter.GetConverter(indexContext);
+
+            var fullyQualifiedResourceId = converterForContext.GetFullyQualifiedResourceId(resource);
+            var serialized = ExpressionSerializer.SerializeExpression(fullyQualifiedResourceId);
+
+            writer.WriteValue(serialized);
+        }
```

Comment:
Unless I'm mistaken, the implementation looks identical to the existing function `EmitResourceIdReference` - can we commonize rather than creating a new one?